src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -389,6 +390,20 @@ public Task PublishUpdateAsync(IResource resource, string resourceId, Func<Custo
         return Task.CompletedTask;
     }
 
+    /// <summary>
+    /// Update resource snapshot health status if the resource is running with no health checks.
+    /// </summary>
+    private static CustomResourceSnapshot UpdateHealthStatus(IResource resource, CustomResourceSnapshot previousState)
+    {
+        // A resource is also healthy if it has no health check annotations and is in the running state.
+        if (!resource.TryGetAnnotationsIncludingAncestorsOfType<HealthCheckAnnotation>(out _) && previousState.State?.Text == KnownResourceStates.Running)
```

Comment:
The new `HasAnnotationIncludingAncestorsOfType<T>` method in https://github.com/dotnet/aspire/pull/6357 would make this cheaper to call.

Reply:
It becomes much cheaper to call if the first condition is `previousState.HealthStatus is not HealthStatus.Healthy` - changed